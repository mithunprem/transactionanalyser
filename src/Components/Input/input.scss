.input {
  position: relative;

  input {
    cursor: text;
    height: 32px;
    box-sizing: border-box;
    padding: 8px;
    font-size: 16px;
    border: none;
    width: 100%;
    box-shadow: none;
    outline: none;
    box-sizing: border-box;
    font-weight: 400;
    background: white;
    margin: 8px 0 16px;
  }
}
