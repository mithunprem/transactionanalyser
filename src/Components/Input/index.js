import React, { Fragment } from 'react';
import { Tooltip } from 'reactstrap';
import PropTypes from 'prop-types';
import './input.scss';

const Input = ({ type = "text", placeholder, name, value, showError, errorMessage, onChange = () => { } }) => {
  return (
    <Fragment>
      <div className="input">
        <label htmlFor={name}>{placeholder}</label>
        <br/>
        <input
          id={name}
          placeholder={placeholder}
          className={showError ? 'has-error' : ''}
          type={type}
          name={name}
          value={value}
          onChange={onChange}
        />
      </div>
      <Tooltip placement="top" target={name} fade={false} isOpen={showError}>{errorMessage}</Tooltip>
    </Fragment>
  )
}

Input.propTypes = {
  type: PropTypes.string,
  placeholder: PropTypes.string,
  name: PropTypes.string,
  showError: PropTypes.bool,
  value: PropTypes.string,
  onChange: PropTypes.func
}

export default Input;
