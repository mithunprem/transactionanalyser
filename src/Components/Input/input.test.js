import React from 'react';
import { create } from 'react-test-renderer';
import Input from '../../Components/Input';

describe("Input", () => {
  const div = document.createElement('div');
  div.setAttribute("id", "input");
  document.body.appendChild(div);

  test("should render without throwing an error", () => {
    const onChange = () => {};
    const input = create(<Input name="input" placeholder="test input" value="test" onChange={onChange}/>).toJSON()
    expect(input).toMatchSnapshot()
  });
});
