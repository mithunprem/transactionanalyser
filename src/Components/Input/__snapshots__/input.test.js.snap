// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should render without throwing an error 1`] = `
<div
  className="input"
>
  <label
    htmlFor="input"
  />
  <br />
  <input
    className=""
    id="input"
    name="input"
    onChange={[Function]}
    placeholder="test input"
    type="text"
    value="test"
  />
</div>
`;
